docs/variants/qemu_q35/development.md

```diff
@@ -125,3 +125,112 @@ System Features. The below image shows the Dasharo System Features enabled for
 the QEMU Q35 machine.
 
 ![DasharoSystemFeatures](images/q35-dasharo-features.jpg)
+
+## Network support
+
+### Compile iPXE for QEMU
+
+To enable iPXE please follow this guide:
+
+```bash
+git clone https://git.ipxe.org/ipxe.git
+```
+
+Change directory:
+
+```bash
+cd ipxe
+```
+
+Checkout stable commit (2023.8 used by coreboot)
+
+```bash
+git checkout 4bffe0f0d9d0e1496ae5cfb7579e813277c29b0f
+```
+
+#### Enabled Dasharo script
+
+```bash
+sed -i 's|//#define\s*IMAGE_SCRIPT.*|#define IMAGE_SCRIPT|' "src/config/general.h"
+```
+
+Get the script:
+
+```bash
+wget https://raw.githubusercontent.com/Dasharo/dasharo-blobs/main/dasharo/dasharo.ipxe
+```
+
+#### Enabled HTTPS
+
+```bash
+sed -i 's|.*DOWNLOAD_PROTO_HTTPS|#define DOWNLOAD_PROTO_HTTPS|g'  "src/config/general.h"
+```
+
+#### Build
+
+Start the instance of the docker image under the Dasharo/edk2 repository:
+
+```bash
+docker run --rm -it -v $PWD/../edk2:/home/coreboot/edk2 \
+    -v $PWD/../edk2-platforms:/home/coreboot/edk2-platforms \
+    -v $PWD:/home/coreboot/ipxe \
+    -w /home/coreboot/ipxe coreboot/coreboot-sdk:2022-12-18_3b32af950d \
+    /bin/bash
+```
+
+Export cross compiler prefix:
+
+```bash
+export CROSS_COMPILE="x86_64-elf-"
+```
+
+Compile:
+
+```bash
+make -C src bin-x86_64-efi-sb/ipxe.efi EMBED=$PWD/dasharo.ipxe BUILD_ID_CMD="echo 0x1234567890" \
+    EXTRA_CFLAGS="-Wno-address-of-packed-member  -m64  -fuse-ld=bfd \
+    -Wl,--build-id=none -fno-delete-null-pointer-checks -Wlogical-op -march=nocona \
+    -malign-data=abi -mcmodel=large -mno-red-zone -fno-pic"
+```
+
+#### Build EDK II
+
+Change directory:
+
+```bash
+cd ../edk2
+```
+
+Setup the environment variables with the following command
+
+```bash
+source edksetup.sh
+```
+
+Compile EDK II base tools:
+
+```bash
+make -C BaseTools
+```
+
+Update the PACKAGES_PATH variable
+
+```bash
+export EDK2_PLATFORMS_PATH="$HOME/edk2-platforms"
+export PACKAGES_PATH="$WORKSPACE:\
+$EDK2_PLATFORMS_PATH/Platform/Intel:\
+$EDK2_PLATFORMS_PATH/Silicon/Intel:\
+$EDK2_PLATFORMS_PATH/Features/Intel:\
+$EDK2_PLATFORMS_PATH/Features/Intel/Debugging:\
+$EDK2_PLATFORMS_PATH/Features/Intel/Network:\
+$EDK2_PLATFORMS_PATH/Features/Intel/OutOfBandManagement:\
+$EDK2_PLATFORMS_PATH/Features/Intel/PowerManagement:\
+$EDK2_PLATFORMS_PATH/Features/Intel/SystemInformation:\
+$EDK2_PLATFORMS_PATH/Features/Intel/UserInterface"
+```
+
+To build the image simply invoke the following command
+
+```bash
+build -a IA32 -a X64 -t GCC5 -b RELEASE -p OvmfPkg/OvmfPkgX64.dsc
```

Comment:
So we have two separate build instructions:
1. Building manual
2. Development

Reply:
@miczyg1 those are almost the same, but Building Manual is not modular enough to scale into this. TBH, I can remove the whole section from here; those were mostly development notes while enabling networking. If that causes trouble in the review, I would rather remove the whole section than waste our time.

Reply:
My concern is that we should rather unify things, not divide and duplicate/repeat (which you also dislike IIRC).

Reply:
@miczyg1 in that case, it would be redesigning the building manual for all supported platforms, which is out of scope here. Knowing how much can be improved, we can easily achieve infinite scope creep - it looks like the removal of old AMD from the top of the coreboot tree. I pointed to inconsistencies in the building manual two years ago: https://github.com/Dasharo/dasharo-issues/issues/37

We had no issues merging tons of documentation even though the problem was present. I wonder why we apply different standards right now.

To satisfy your comment, I will remove that section from the document. I'm concerned about the development section at all, but it was approved at some point, so I guess it meets your or someone else quality criteria. Fix:

Let's also think about how we can accelerate time to market - meaning release delivery instead of delaying it forever by repeatedly nitpicking. I recommend reading [this](https://blog.danlew.net/2021/02/23/stop-nitpicking-in-code-reviews/) because perfect is often the enemy of good.

Reply:
I guess we still don't understand each other. My point its:

Why do we need separate build instructions for development and build manual?
 
If we want to achieve the same functionality iPXE should be always built, not matter dev or not, so... Includig iPXE build procedure in build manual should be the right thing to do. Build manual always describes the procedure to build the same binaries as the release binaries. The same should apply here. 

> We had no issues merging tons of documentation even though the problem was present. I wonder why we apply different standards right now.
 
No idea what you mean. I simply want to follow the same standards and say that you don't need separate build instructions...

Reply:
> I guess we still don't understand each other here @pietrushnic . My point its:

@miczyg1 definitely.

> Why do we need separate build instructions for development and build manual?

We don't. I removed separate building instructions in the commit I pointed to https://github.com/Dasharo/docs/commit/8a5e23cfbd71728179605db9d735c229852e2ed0

> If we want to achieve the same functionality iPXE should be always built, not matter dev or not, so... Includig iPXE build procedure in build manual should be the right thing to do. Build manual always describes the procedure to build the same binaries as the release binaries. The same should apply here.

This is precisely the state of the commitment right now. Please point to the precise line which should be changed, because you discuss something that does not exist now.

> No idea what you mean. I simply want to follow the same standards and say that you don't need separate build instructions...

Previously you wrote:

> My concern is that we should rather unify things, not divide and duplicate/repeat (which you also dislike IIRC).

So please take a look at the building manuals. Those are not unified. If you want to avoid dividing, duplicating/repeating, please do not merge building manual instructions that are not unified. IMO, in this discussion, different rules are applied because previously not, unified documentation was merged, but now when we had some duplication (one line, what was already fixed), we apply some different rules than the unification rule you explain. I just point out inconsistencies in the review process. I will try to address those in a better suited place than this thread. Let's focus on the code.

Reply:
> So please take a look at the building manuals. Those are not unified. 

Yes they aren't. But we can slowly work towards that in separate efforts elsewhere. ANyways, my point was addressed to not duplicate building instructions in the same platform boundary. Closing.